workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -135,8 +135,28 @@ def _restore_files_to_manager(self):
 
     def _restore_db(self, postgres):
         ctx.logger.info('Restoring database')
-        if self._snapshot_version >= V_4_0_0:
+        if self._snapshot_version > V_4_0_0:
+            postgres.restore(self._tempdir)
+        elif self._snapshot_version == V_4_0_0:
+            base_script = (
+                'import flask_migrate;'
```

Comment:
I've just committed a refactoring that basically moves the ugly code to the utils module. Please let me know if it looks better now.